src/main/java/uk/gov/di/utils/CoreIdentityValidator.java

```diff
@@ -98,7 +102,8 @@ private void verifyController(String controller, VerificationMethod verification
                             + verificationMethod.getController().toString());
         }
         var expectedControllerFromHost = "did:web:" + didKeyUri.getAuthority();
-        if (!controller.equals(expectedControllerFromHost)) {
+        if (!controller.replace("%3A", ":").equals(expectedControllerFromHost)) {
+            LOG.info(expectedControllerFromHost);
             throw new RuntimeException(
                     "Controller in User Identity kid does not match DID key URL: "
                             + controller
```

Comment:
Do we perhaps want to throw using the replaced string too?

Reply:
Yeah probably!